contrib/bindings/go/pathrs/pathrs.go

```diff
@@ -0,0 +1,328 @@
+// Copyright (C) 2019, 2020 Aleksa Sarai <cyphar@cyphar.com>
+// Copyright (C) 2020 Maxim Zhiburt <zhiburt@gmail.com>
+// Copyright (C) 2019, 2020 SUSE LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package pathrs provides a bindings for libpathrs, a library for safe path resolution on Linux.
+package pathrs
+
+// #cgo LDFLAGS: -lpathrs
+// #include <pathrs.h>
+import "C"
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"syscall"
+	"time"
+	"unsafe"
+)
+
+// Root holds the responsibility to provide safe api to functions of pathrs root api
+type Root struct {
+	root *C.pathrs_root_t
+}
+
+// Handle represents an handle pathrs api interface
+type Handle struct {
+	handle  *C.pathrs_handle_t
+	nameGen *rand.Rand
```

Comment:
There's no real need for this -- the top-level `"math/rand".Read` is safe for concurrent access (and in fact, by storing a `rand.Rand` this randomness source is no longer safe for concurrency). Another alternative is `"crypto/rand".Read`.

Reply:
Also (sorry, I should've been clearer about this in my last review) this `Handle` declaration should be right above where you define the methods for `*Handle` -- as is common with most Go codebases.

Reply:
Correct but we need `rand.Seed`. I placed that here since I am not honestly confident that calling `rand.Seed`, which is thread-safe for sure, is better since it affects others by changing the global state. Anyway, at least it does it safely :).

Reply:
`"crypto/rand".Read` doesn't need a seed.

Reply:
I did not pointe that feature out, sorry. I will stick with "crypto/rand".Read